src/edu/washington/escience/myria/operator/Split.java

```diff
@@ -0,0 +1,146 @@
+import edu.washington.escience.myria.storage.TupleBatch;
+import edu.washington.escience.myria.storage.TupleUtils;
+
+/**
+ * Generic apply operator.
+ */
+public class Split extends UnaryOperator {
+  /***/
+  private static final long serialVersionUID = 1L;
+
+  /**
+   * Name of column to split on regex. Must have String type.
+   */
+  private final String columnName;
+
+  /**
+   * Compiled regex to split input tuples.
+   */
+  private final Pattern pattern;
+
+  /**
+   * Schema of output tuples.
+   */
+  private Schema outputSchema;
+
+  /**
+   * Index of column to split in input schema (and anonymous result column in output schema).
+   */
+  private int splitColIdx;
+
+  /**
+   * Array of column indices in output schema corresponding to column indices of input schema. The column to be split
+   * has its index mapped to the sentinel value -1.
+   */
+  private int[] inputColIndexToOutputColIndex;
+
+  /**
+   * 
+   * @param child child operator that data is fetched from
+   * @param columnName name of string column to split using regex
+   * @param regex regular expression to split string input column on
+   */
+  public Split(final Operator child, final String columnName, final String regex) {
+    super(child);
+    this.columnName = columnName;
+    pattern = Pattern.compile(regex);
+  }
+
+  @Override
+  protected TupleBatch fetchNextReady() throws DbException {
+    final Operator child = getChild();
+    for (TupleBatch tb = child.nextReady(); tb != null; tb = child.nextReady()) {
+      List<ColumnBuilder<?>> colBuilders = new ArrayList<>(tb.numColumns());
+      for (int i = 0; i < tb.numColumns(); ++i) {
+        if (i != splitColIdx) {
+          colBuilders.add(ColumnFactory.allocateColumn(tb.asColumn(i).getType()));
+        }
+      }
+      final ColumnBuilder<?> splitColBuilder = ColumnFactory.allocateColumn(Type.STRING_TYPE);
+      colBuilders.add(splitColBuilder);
+      for (int rowIdx = 0; rowIdx < tb.numTuples(); ++rowIdx) {
+        try {
+          String colValue = tb.asColumn(splitColIdx).getString(rowIdx);
+          // We must specify a negative value for the limit parameter to avoid discarding trailing empty strings:
+          // http://docs.oracle.com/javase/7/docs/api/java/lang/String.html#split(java.lang.String,%20int)
+          String[] splits = pattern.split(colValue, -1);
+          for (String segment : splits) {
+            splitColBuilder.appendString(segment);
+            // For each split segment, duplicate the values of all other columns in this row.
+            for (int inputColIdx = 0; inputColIdx < inputColIndexToOutputColIndex.length; ++inputColIdx) {
+              int outputColIdx = inputColIndexToOutputColIndex[inputColIdx];
+              if (outputColIdx != -1) {
+                TupleUtils.copyValue(tb.asColumn(inputColIdx), rowIdx, colBuilders.get(outputColIdx));
+              }
+            }
+          }
+        } catch (Exception e) {
```

Comment:
We don't do blanket `Exception` in Myria. http://www.javaworld.com/article/2077500/testing-debugging/java-tip-134--when-catching-exceptions--don-t-cast-your-net-too-wide.html Did you have a reason for doing so here?

What are the actual expected exceptions?